### ISSUE TITLE
Model Incorporation

### ISSUE THREAD

**Issue by pittmanriley** [ersilia-os/eos2hzy#1]
Hi @GemmaTuron, I've gotten this model working, but I'm a little bit confused by the interpretation of the output of the model. It says in the description that it "looks for similar molecules to the input molecule and returns a list of 1000 molecules by default." If I test this model from its original folder, the chem-sampler folder, using the command `python src/sampler.py 'data/example_drugs.csv' 'data/output.csv' --sampler PubChemSampler --num_samples 100 --sim_ub 0.9 --sim_lb 0.3 --distribution ramp`, it prints 1000 similar molecules to the console, but in the output folder specified, it only shows 24. Attached here are the data/example_drugs.csv (input) and data/output.csv (output) files so you can see:
[example_drugs.csv](https://github.com/ersilia-os/eos2hzy/files/12375888/example_drugs.csv)
[output.csv](https://github.com/ersilia-os/eos2hzy/files/12375890/output.csv)

Looking at this, I'm getting confused because it's not really clear what the output is for each molecule from the input example_drugs.csv file. The output just shows a single column of molecules, and it doesn't show which input the molecules are similar to. Is it supposed to be like this? Or is this a mistake? I can't tell if the model is outputting molecules similar to each input, or if it outputs a list of molecules that are similar to every single one of the inputs. Perhaps this is the correct interpretation? 

Let me know if my question doesn't make sense. Thanks!

**Comment by GemmaTuron** [comment]
Hi @pittmanriley 

Indeed, in this case the model is prepared to take all smiles at once and give back a list (if you see it, it is actually even shuffled at the end of the pipeline) - we can discuss if we want it smiles by smiles, I think in this case it would make sense to do so, basically get each smiles and their similar molecules, fixing the output to 100 for the moment (so, a `List`) 

You basically just need to play a bit with the sample function to iterate over each smiles and collect the results separately, if this makes sense?

**Comment by pittmanriley** [comment]
@GemmaTuron this makes sense! I’ll play around with it today and this weekend.

**Comment by pittmanriley** [comment]
Hi @GemmaTuron, I wasn't able to get very far over the weekend. I changed the output in the code to make it so that it shows the similar molecules for each input smiles, which works when I run using bash. But, I'm getting errors when I run it after fetching from --repo_path, which I assume means there's some issue with how the service.py file is processing the output. I'm not sure how much time I'll be able to keep dedicating to this. How should I proceed? Should I keep working on it in my free time, or just call it a day and push what I've done?
